### ISSUE TITLE
Task – Assignee with the custom name is shown as email in a task name

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#36443]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.41-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:**

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Login
3. Navigate to a Group Chat
4. Click on the + in the compose box and select "Assign task"
5. Enter a a Title
6. Assign a user with the custom name
7. Create the task

## Expected Result:
The assignee with the custom name is displayed in the assignee field with custom name

## Actual Result:
Assignee with the custom name is shown as email in a task name

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/154eefbc-a667-4785-a186-923a8832077d

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
We think that this bug might be related to #vip-vsb
CC @quinthar

**Comment by jeremy-croff** [comment]
Regression?
https://github.com/Expensify/App/commit/8c802edccf7f9221e06508086db22e94ba806afd

**Comment by jjcoffee** [comment]
Thanks @jeremy-croff, that does look like a possible regression. Checking with the [PR](https://github.com/Expensify/App/pull/35027#issuecomment-1943390876) author.

**Comment by jjcoffee** [comment]
Looks like it's actually a regression from https://github.com/Expensify/App/pull/35462 as the order was [changed here](https://github.com/Expensify/App/pull/35462/files#diff-a4fc8bbb64960d70ffb3c3d2b6dae62c6881855d6e3c66d9e3b5976441832b3dR42). @mollfpr Can you check?

**Comment by mollfpr** [comment]
@jjcoffee Yeah, I think you are right. It should be prioritizing the `displayName`. Sorry, I didn't notice it, because the old file was deleted instead of renaming it 🙏

**Comment by jjcoffee** [comment]
No worries, those are hard to catch! @ruben-rebelo As you worked on the PR, are you able to raise a quick fix for this regression?

**Comment by paultsimura** [comment]
Hey @ruben-rebelo @mollfpr just another bump on this – we need the order to be fixed for another task as well 🙂

**Comment by jjcoffee** [comment]
@adelekennedy This is a regression from https://github.com/Expensify/App/pull/35462 but this is blocking [other](https://github.com/Expensify/App/issues/36443#issuecomment-1952255312) issues being fixed, so do you reckon it's worth just progressing with this as a normal issue, rather than waiting on @ruben-rebelo to raise a fix?

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The login in mentions takes priority over the display name.

### What is the root cause of that problem?
This is a regression from https://github.com/Expensify/App/pull/35462, where the order of preference was changed:

https://github.com/Expensify/App/blob/3ce5246ae44565e12aae6139c529d61ba2e6920e/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx#L42

### What changes do you think we should make in order to solve the problem?
We need to change the order of the display name calculation back to how it was pre-migration:

```js
displayNameOrLogin = user?.displayName || LocalePhoneNumber.formatPhoneNumber(user?.login ?? '') || translate('common.hidden');
```

### What alternative solutions did you explore? (Optional)

**Comment by jjcoffee** [comment]
Happy to go with @paultsimura's [proposal](https://github.com/Expensify/App/issues/36443#issuecomment-1952636024) here in order to get this unblocked, as it doesn't seem that (regression PR author) @ruben-rebelo is working on a fix.

:ribbon::eyes::ribbon: C+ reviewed

**Comment by paultsimura** [comment]
Thanks. The PR is ready for review: https://github.com/Expensify/App/pull/36998

**Comment by adelekennedy** [comment]
SORRY - late payment from me

Payouts due:

Contributor: $500 @paultsimura (Upwork)
Contributor+: $500 @jjcoffee (Upwork)

Upwork job is [here](https://www.upwork.com/jobs/~01a831cc646a8c8645).

**Comment by adelekennedy** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] The PR that introduced the bug has been identified. Link to the PR:
- [ ] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the GH issue for creating the test here:

**Comment by jjcoffee** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/35462
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/35462/files#r1521218238
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A**
- Determine if we should create a regression test for this bug. **Yes - if one doesn't exist already**
- If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

1. Create a task with any title
2. Assign a user with a display name set
3. Share somewhere
4. Navigate to the chat where the task was shared
5. Verify the assignment in the task preview shows the user's custom name

Do we agree 👍 or 👎

**Comment by adelekennedy** [comment]
makes sense to me - I'll wait if @techievivek has any more input

**Comment by jjcoffee** [comment]
@adelekennedy I think this can be closed?

**Comment by techievivek** [comment]
Sorry I missed the ping, I have created a GH to add the regression test here: https://github.com/Expensify/Expensify/issues/385997, I am guessing payment is done here so we can close it out.
